clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -105,6 +105,10 @@ public final class ConsumerCoordinator extends AbstractCoordinator {
     private final boolean autoCommitEnabled;
     private final int autoCommitIntervalMs;
     private final ConsumerInterceptors<?, ?> interceptors;
+    // track number of async commits for which callback must be called
+    // package private for testing
+    final AtomicInteger inFlightAsyncCommits;
+    // track number of inflight coordinator lookups
```

Comment:
nit: Could we say `track the number of pending async commits waiting on the coordinator lookup to complete`.?

Reply:
Of course! Done in https://github.com/apache/kafka/pull/13678/commits/e4ba315ecbf0878706de1e1b620911896a994bed